### DISCUSSION TITLE
Problem adding river networks from SSN2

### DISCUSSION THREAD

**Discussion by ErikStrandh** [davidbolin/MetricGraph#12] [Q&A] [answered]
Hi!

When trying to create a graph of a river network from the SSN2 package as in the vignette,

`library(SSN2)

copy_lsn_to_temp()
path <- file.path(tempdir(), "MiddleFork04.ssn")

mf04p <- ssn_import(
  path = path,
  predpts = c("pred1km", "CapeHorn"),
  overwrite = TRUE
)

riverGraph <- MetricGraph::metric_graph$new(mf04p)`

I get the following error message,

Starting graph creation...
LongLat is set to FALSE
The current tolerances are:
         Vertex-Vertex 0.001
         Vertex-Edge 0.001
         Edge-Edge 0
Error in check_lines_input(edges) : 
  The list must contain either matrices of data.frames!

I was able to fix it by, 
`library(sf)

edges <- sf::st_multilinestring(mf04p$edges$geometry[,2])
riverGraph <- MetricGraph::metric_graph$new(edges =  edges)`

but this does not seem optimal and I have not been able to add all of the data to the graph. Is there an easy fix?

Best regards

**Comment by davidbolin** [comment, marked as answer]
Hi, are you using the devel version? The CRAN version is outdated by now and will soon be updated.

**Reply by ErikStrandh** [reply]
Yes, used the CRAN version. Downloaded the devel version and now the code from the vignette works.

Thank you for the quick answer.
